qa/tasks/cephfs/test_uninlining.py

```diff
@@ -0,0 +1,323 @@
+                    json_out = self.fs.rank_tell(['dump', 'inode', inode], rank)
+                    return json_out
+                except json.decoder.JSONDecodeError:
+                    time.sleep(1)
+            raise json.decoder.JSONDecodeError(f'No JSON found after {retries} attempts', None, 0)
+
+        info = []
+        for i in range(len(inodes)):
+            inode = inodes[i]
+            log.debug(f"getting inode info #{i+1} of {len(inodes)}:{inode}")
+            path = os.path.dirname(files[i])
+            rank = dir_pins[path] if path in dir_pins else 0
+            r = rank
+            while r < rank + num_mds:
+                try:
+                    json_out = get_inode_dump(inode,
+                                              r % num_mds,
+                                              self.DUMP_INODE_RETRIES)
+                    break
+                except json.decoder.JSONDecodeError:
+                    pass
+                finally:
+                    r += 1
+            self.assertTrue(json_out is not None)
+            self.assertTrue('inline_data_length' in json_out)
+            self.assertTrue('inline_data_version' in json_out)
+            info.append(InlineDataInfo(json_out['inline_data_length'],
+                                       json_out['inline_data_version']))
+        return info
+
+    def run_test_worker(self,
+                        opt_clear_cache,
+                        opt_unmount,
+                        opt_mount_openbg,
+                        opt_multimds,
+                        opt_snapshot):
+        log.info("Running Data Uninlining test with: "
+                 f"{self.cache_info[opt_clear_cache]}, "
+                 f"{self.unmount_info[opt_unmount]}, "
+                 f"{self.mount_openbg_info[opt_mount_openbg]}, "
+                 f"{self.multimds_info[opt_multimds]}, "
+                 f"{self.snapshot_info[opt_snapshot]}")
+
+        # Set max_mds to 1 or 2
+        num_mds = 2 if opt_multimds else 1
+        log.debug(f"setting max_mds:{num_mds}")
+        self.fs.set_max_mds(num_mds)
+
+        # Get configured max inline data size
+        log.debug("getting client_max_inline_size")
+        idsize = self.fs.fs_config.get('client_max_inline_size', 4096)
+        idsize = int(idsize)
+        log.debug(f"got client_max_inline_size:{idsize}")
+
+        snapshotter = None
+        if opt_snapshot:
+            log.debug("starting snapshotter thread")
+            cmd = 'mkdir linux-5.4'
+            remote_mntpt_cmd(self.mount_b, cmd)
+            snapshotter = SnapshotterThread("linux-5.4",
+                                            self.NUM_SNAPS,
+                                            self.mount_b)
+            snapshotter.start()
+
+        # Extract test data tarball
+        # FIXME
+        log.debug("extracting tarball")
+        cmd = 'tar -x -z -f linux-5.4.tar.gz linux-5.4/fs/ceph linux-5.4/fs/orangefs linux-5.4/fs/ext2'
```

Comment:
BTW, shouldn't we remove the `linux-5.4/` to make it clean for each test ? Doesn't it matter here ?

Reply:
Okay, please ignore this, I saw you have done this.

Reply:
The `linux-5.4` dir is removed on line 261 after each iteration.